test/k8sT/DatapathConfiguration.go

```diff
@@ -405,10 +406,14 @@ var _ = Describe("K8sDatapathConfig", func() {
 			deleteETCDOperator(kubectl)
 		})
 		It("Check connectivity with managed etcd", func() {
-			deployCilium([]string{
+			opts := []string{
 				"--set global.etcd.enabled=true",
 				"--set global.etcd.managed=true",
-			})
+			}
+			if os.Getenv("NO_CILIUM_ON_NODE") != "" {
+				opts = append(opts, "--set operator.synchronizeK8sNodes=false")
+			}
+			deployCilium(opts)
```

Comment:
`cilium-operator` adds the nodes into the kvstore so that all `cilium-agents` will watch for kvstore events. If we stop doing this won't we potentially brake tests that depend on that behavior?

Reply:
This was breaking the `ManagedEtcd` test case, as `cilium-operator` was treating `k8s3` as a cilium node.

Each cilium-agent announces itself by pushing CiliumNode object. I haven't seen any breaking test due to missing cilium node in a cluster.